gcm_toolkit/utils/interface.py

```diff
@@ -487,6 +487,156 @@ def phase_curve(self, phases, spectra=None, filename=None):
             coords={"phase": phases, "wlen": spectra.wlen},
         )
 
+    def calc_transit_spectrum(
+            self,
+            mmw,
+            lat_points=4,
+            lon_resolution=10,
+            gravity=None,
+            rplanet=None,
+            pressure_0=None,
+            co_ratio=0.55,
+            feh_ratio=0.0,
+            mass_frac=None,
+    ):
+        """
+        Calculate the transit spectrum. This function avarages T-p profiles in
+        the terminator region and uses poorman equilbriums chemistry.
+
+        Parameters
+        ----------
+        mmw: float or 1D-array
+            Mean molecular weight (in atomic units). Will be globally uniform if
+            float or horizonatally uniform if 1D.
+        lat_points : int
+            Number of latitude points considered per terminator.
+        lon_resolution : float
+            Longitude angle to avarage over at the equator of the terminators. In !degrees!.
+        gravity: float
+            surface gravity in !cgs!. Will default to the value provided by GCMT.
+        rplanet: float
+            planet radius in !cm!. Will default to the value provided by GCMT.
+        pressure_0: float
+            pressure level of the gravity and radius. Will default to the value provided by GCMT.
+        co_ratio: float, optional
+            The C/O ratio. Currently only one global value allowed. Defaults to 0.55.
+        feh_ratio: float, optional
+            The metalicity ratio. Currently only one global value allowed.
+            Defaults to 0.0.
+        mass_frac: List[List[Dict]], optional
+            The mass fractions for each t_p point. Structure is as follows: mass_frac[i][j]['key']
+            - i: [2 elements] 0 for morning terminator, 1 for evening terminator
+            - j: [lat_points elements] latitude point starting from lowest (closeset to lat = -90) to highest (closest to lat = 90)
+            - dict must contain all mass fractions  elements given as opacity species to prt
+
+
+        Returns
+        -------
+        spectra: xr.DataArray
+            Dataarray containing the spectrum.
+            The spectrum is normed to the stellar spectrum.
+
+        """
+        from petitRADTRANS.poor_mans_nonequ_chem import interpol_abundances
```

Comment:
I would prefer to use `self.chemistry` instead. The pRT interface is initialized at some point with a chemistry scheme, which can be any chemistry scheme.

You then get the abundances via 
```py
        abus = self.chemistry.to_prt(
            self.prt.line_species, self.prt.press / 1e6
        )
```

Have a look at `calc_phase_spectrum`. Here is how `abus` are used there:
```py
        for i, lon_i in enumerate(np.array(lon).flat):
            lat_i = np.array(lat).flat[i]
            theta_list.append(theta_star.sel(lon=lon_i, lat=lat_i).values)
            temp_list.append(abus[c["T"]].sel(lon=lon_i, lat=lat_i).values)
            abunds_list.append(
                {
                    sp: abus[sp].sel(lon=lon_i, lat=lat_i).values
                    for sp in prt_abu_keys
                }
            )
```

You could adapt it to use the provided mmr dict, if there is one.

Reply:
see below